pkg/controller/resourceclaim/metrics/metrics.go

```diff
@@ -45,6 +45,22 @@ var (
 			Help:           "Number of ResourceClaims creation request failures",
 			StabilityLevel: metrics.ALPHA,
 		})
+	// NumResourceClaims tracks the current number of ResourceClaims.
+	NumResourceClaims = metrics.NewGauge(
+		&metrics.GaugeOpts{
+			Subsystem:      ResourceClaimSubsystem,
+			Name:           "num_resource_claims",
```

Comment:
nit: `num` is implicit with `resource_claims`, I think `resourceclaim_controller_resource_claims` would be a slightly better name

same for the other metric.

Reply:
`resourceclaim_controller` is already part of the full name because `ResourceClaimSubsystem` has that string:

https://github.com/kubernetes/kubernetes/blob/b736a4e75f5538910135db7f4329e634ac221748/pkg/controller/resourceclaim/metrics/metrics.go#L27

I picked `num_resource_claims` for consistency with `endpoint_slice_controller_num_endpoint_slices`:

https://github.com/kubernetes/kubernetes/blob/d99d3f7eb734dc9284a90374fcc6d8cc32d1946d/pkg/controller/endpointslicemirroring/metrics/metrics.go#L103-L107

I don't have a strong opinion on this.

Reply:
Ack. I've removed "num_" from this PR and the KEP update.